test/MarginAccount.test.ts

```diff
@@ -0,0 +1,485 @@
+      //   await expectRevert(marginAccountTester.testAddCollateral(otoken.address, 10, 2))
+      //   await expectRevert(marginAccountTester.testAddCollateral(otoken.address, 10, 1))
+      //   await expectRevert(marginAccountTester.testAddCollateral(otoken.address, 10, 0))
+      // })
+      //
+      // it('should revert if trying on the wrong vault', async () => {
+      //   // await expectRevert(marginAccountTester.testAddShort(otoken.address, 10, 5, {from: deployer}), 'revert')
+      //   // await expectRevert(await marginAccountTester.testAddShort(Otoken, 10, 3, {from: deployer}), 'revert')
+      //   // await expectRevert(await marginAccountTester.testAddShort(Otoken, 10, 100, {from: deployer}), 'revert')
+      // })
+    })
+
+    describe('Remove short', () => {
+      before('redeploying', async () => {
+        marginAccountTester = await MarginAccountTester.new()
+      })
+
+      // testAccount = {owner: ZERO_ADDR, vaultIds: new BigNumber(0)}
+      it('vaultIds should be zero again, owner should be ZERO_ADDR', async () => {
+        account = await marginAccountTester.getAccount({from: deployer})
+        assert.equal(account.owner, testAccount.owner, 'MarginAccount.Account owner addr mismatch')
+        assert.equal(account.vaultIds, testAccount.vaultIds, 'MarginAccount.Account owner vaultids mismatch')
+      })
+
+      it('should not remove short otokens if there are none', async () => {
+        await expectRevert(
+          marginAccountTester.testRemoveShort(otoken.address, 10, 0),
+          'MarginAccount: short otoken address mismatch',
+        )
+      })
+
+      it('should add some short otokens to be removed later', async () => {
+        await marginAccountTester.testAddShort(otoken.address, 10, 0)
+      })
+
+      it('should remove some of those short otokens', async () => {
+        const index = 0
+        const toRemove = 5
+        const beforeVault = await marginAccountTester.getVault()
+        await marginAccountTester.testRemoveShort(otoken.address, toRemove, index)
+        const afterVault = await marginAccountTester.getVault()
+        assert.equal(
+          new BigNumber(beforeVault.shortAmounts[index])
+            .minus(new BigNumber(afterVault.shortAmounts[index]))
+            .toString(),
+          new BigNumber(toRemove).toString(),
+          'amount removed mismatch',
+        )
+      })
+
+      it('should not be able to remove more than the num of remaining of those short otokens', async () => {
+        await expectRevert(
+          marginAccountTester.testRemoveShort(otoken.address, 100, 0),
+          'SafeMath: subtraction overflow',
+        )
+      })
+
+      //TODO: Is this a condition we need?
+      // it('should not be able to remove zero of those short otokens', async () => {
+      //   await marginAccountTester.testRemoveShort(otoken.address, 0, 1)
+      //   const vault = await marginAccountTester.getVault()
+      //   assert.equal(1,2)
+      // })
```

Comment:
@antoncoding @haythem96 thoughts?

Reply:
Can u explain more this test ?

Reply:
If there's no `amount != 0` constraint in the MarginAccount contract then this is not necessary. But not bad to have it.